cartographer/mapping_3d/sparse_pose_graph.cc

```diff
@@ -433,44 +455,35 @@ std::vector<transform::Rigid3d> SparsePoseGraph::ExtrapolateSubmapTransforms(
   if (trajectory_ids_.count(trajectory) == 0) {
     return {transform::Rigid3d::Identity()};
   }
-  const int trajectory_id = trajectory_ids_.at(trajectory);
-  size_t flat_index = 0;
-  size_t flat_index_of_result_back = -1;
-
-  // Submaps for which we have optimized poses.
-  std::vector<transform::Rigid3d> result;
-  for (; flat_index != submap_states_.size(); ++flat_index) {
-    const auto& state = submap_states_[flat_index];
-    if (state.id.trajectory_id != trajectory_id) {
-      continue;
-    }
-    if (static_cast<size_t>(trajectory_id) >= submap_transforms.size() ||
-        result.size() >= submap_transforms.at(trajectory_id).size()) {
-      break;
-    }
-    result.push_back(
-        submap_transforms.at(trajectory_id).at(result.size()).pose);
-    flat_index_of_result_back = flat_index;
+  const size_t trajectory_id = trajectory_ids_.at(trajectory);
+  if (trajectory_id >= submap_states_.size()) {
```

Comment:
unfortunately, this is possible since trajecotry_ids_ grows also when sensor data is added (for example IMU), but submap_states_ not.